src/server/index.ts

```diff
@@ -88,43 +97,84 @@ function startServer() {
         res.status(404).send(`{"error":"${err}}`);
         return;
       }
-      const data = <JoinGameResponse>{
-        roomURL: game.dailyRoomURL,
-        gameName: game.name,
-        wordSet: game.wordSet,
-      };
-      res.send(data);
+
+      // Check if the user joining has a valid game host cookie
+      const isHost = isGameHostFromSignedCookies(
+        req.signedCookies,
+        game.id,
+        game.createdAt
+      );
+
+      // If they're not a host, just return normal game data
+      if (!isHost) {
+        const data = <JoinGameResponse>{
+          roomURL: game.dailyRoomURL,
+          gameName: game.name,
+          wordSet: game.wordSet,
+        };
+        res.send(data);
+        return;
+      }
+
+      // If they are a host, retrieve a Daily meeting token and
+      // include it in the response.
+      getMeetingToken(game.dailyRoomName)
+        .then((token) => {
+          const data = <JoinGameResponse>{
+            roomURL: game.dailyRoomURL,
+            gameName: game.name,
+            wordSet: game.wordSet,
+            meetingToken: token,
+          };
+          res.send(data);
+        })
+        .catch((error) => {
+          console.error("failed to get meeting token", error);
+          res.sendStatus(500);
+        });
     });
   });
 
   // /create endpoint handles creating a game
   app.post("/create", (req: Request, res: Response) => {
     const body = <CreateGameRequest>req.body;
-    const { wordSet, gameName } = body;
+    const { wordSet, gameName, playerName } = body;
     if (!wordSet) {
       const err = "word set must be defined";
-      res.status(400).send(`{"error":"${err}}`);
+      res.status(400).send(`{"error":"${err}"}`);
       return;
     }
-    if (!gameName) {
-      const err = "game name must be defined";
-      res.status(400).send(`{"error":"${err}}`);
+    for (let i = 0; i < wordSet.length; i += 1) {
+      const word = wordSet[i];
+      const val = word.value;
+      if (!isValidWord(val)) {
+        const err = new InvalidWord(val);
+        res.status(400).send(`{"error":"${err}"}`);
+        return;
+      }
+    }
+    if (!gameName || !isValidName(gameName)) {
+      const err = new InvalidName(gameName);
+      res.status(400).send(`{"error":"${err}"}`);
+      return;
+    }
+    if (!playerName || !isValidName(playerName)) {
+      const err = new InvalidName(playerName);
+      res.status(400).send(`{"error":"${err}"}`);
       return;
     }
+
     orchestrator
       .createGame(gameName, wordSet)
       .then((game) => {
-        getMeetingToken(game.dailyRoomName)
```

Comment:
Removing meeting token retrieval here, because we will now do it on join if the user has a valid game host cookie.